projects/plugins/jetpack/extensions/blocks/subscriptions/panel.js

```diff
@@ -285,6 +287,13 @@ function NewsletterPostPublishSettingsPanel( {
 					</div>
 				</PluginPostPublishPanel>
 			) }
+
+			{ /*
+			     Only show the following panel on WPCOM sites since the
+			     Fees and "Upgrade plan" links displayed are WPCOM specific.
+			  */ }
+
+			{ ! isJetpackSite && <NewsletterPostPublishPanelUpgradeNude /> }
```

Comment:
Do we need to display it when someone hasn't restricted the post to subscribers and / or hasn't even set things up for payments? This way they wouldn't get confused by a message that doesn't apply to them.

Reply:
Great point, Jeremy. I'll only show this panel when:

1. The Paid subscribers option is selected
2. Stripe is connected
3. Newsletter paid plan has been created

Reply:
Just noting there that checking for "The Paid subscribers option is selected" is enough, as it can only be selected when 2. and 3. are already true